core/org.osate.xtext.aadl2.properties/src/org/osate/xtext/aadl2/properties/util/GetProperties.java

```diff
@@ -864,11 +864,11 @@ public static double getMinimumTimeToTransferData(final NamedElement bus, Classi
 	}
 
 	public static double fromMStoSec(NamedElement ne, double value) {
-		return convertToScale(value, getMSUnitLiteral(ne), getSecUnitLiteral(ne));
+		return getMSUnitLiteral(ne) != null ? convertToScale(value, getMSUnitLiteral(ne), getSecUnitLiteral(ne)) : -1;
```

Comment:
I'm not sure that returning -1 is the right thing to do here. Under normal conditions, I would expect that `getMSUnitLiteral()` would never return `null`. If `ms` cannot be found, then something very serious has gone wrong. It is probably better to throw a `NullPointerException` which complains that the unit literal could not be found.